src/popup/components/TransactionOverview.vue

```diff
@@ -83,44 +82,51 @@ export default defineComponent({
         callerId,
       } = innerTx.value;
 
-      if (outerTxTag.value === Tag.PayingForTx && innerTxTag.value === Tag.GaAttachTx) {
-        return {
-          sender: {
-            address: innerTx.value.ownerId,
-            name: getName(innerTx.value.ownerId).value,
-            url: protocolExplorer.prepareUrlForAccount(innerTx.value.ownerId),
-            label: t('multisig.multisigVault'),
-          },
-          recipient: {
-            label: t('common.smartContract'),
-            address: innerTx.value.contractId,
-          },
-        };
-      }
-
-      const transactionTag = (
-        outerTxTag.value === Tag.PayingForTx
-        || outerTxTag.value === Tag.GaMetaTx
-      ) ? innerTxTag.value : outerTxTag.value;
+      // Handle Aeternity-specific transaction types
+      if (protocol.value === PROTOCOLS.aeternity) {
+        if (outerTxTag.value === Tag.PayingForTx && innerTxTag.value === Tag.GaAttachTx) {
+          return {
+            sender: {
+              address: innerTx.value.ownerId,
+              name: getName(innerTx.value.ownerId).value,
+              url: protocolExplorer.prepareUrlForAccount(innerTx.value.ownerId),
+              label: t('multisig.multisigVault'),
+            },
+            recipient: {
+              label: t('common.smartContract'),
+              address: innerTx.value.contractId,
+            },
+          };
+        }
 
-      switch (transactionTag) {
-        case Tag.SpendTx:
+        if (outerTxTag.value === Tag.GaMetaTx) {
           return {
             sender: {
-              address: senderId,
-              name: getName(senderId).value,
-              url: protocolExplorer.prepareUrlForAccount(senderId),
-              label: t('transaction.overview.accountAddress'),
+              address: innerTx.value.gaId,
+              name: getName(innerTx.value.gaId).value,
+              url: protocolExplorer.prepareUrlForAccount(innerTx.value.gaId),
+              label: t('multisig.multisigVault'),
             },
             recipient: {
               address: recipientId,
-              name: name.value || getName(recipientId).value,
```

Comment:
This is the only place in this file where the `name` variable was used. It seems new logic is not resolving name hashes into name.